FAIL: TestDebugLinePrologueParser

Running tests @ https://github.com/derekparker/delve/commit/04c4b019f7557344910ade621f68e2695eabe2e5 on Windows 10 with Go 1.9 gives this:

```
--- FAIL: TestDebugLinePrologueParser (0.98s)
panic: Could not parse string [recovered]
        panic: Could not parse string

goroutine 22 [running]:
testing.tRunner.func1(0xc0420a80f0)
        C:/Go1.9/src/testing/testing.go:719 +0x2e3
panic(0x5537e0, 0x5acea0)
        C:/Go1.9/src/runtime/panic.go:491 +0x291
github.com/derekparker/delve/pkg/dwarf/util.ParseString(0xc042190070, 0xc042033cb8, 0x534099, 0x646460)
        D:/go/src/github.com/derekparker/delve/pkg/dwarf/util/util.go:77 +0x8f
github.com/derekparker/delve/pkg/dwarf/line.parseIncludeDirs(0xc0420d3980, 0xc042190070)
        D:/go/src/github.com/derekparker/delve/pkg/dwarf/line/line_parser.go:96 +0x39
github.com/derekparker/delve/pkg/dwarf/line.Parse(0xc042194000, 0x10600, 0x10600, 0xc042194000, 0x10600, 0x10600)
        D:/go/src/github.com/derekparker/delve/pkg/dwarf/line/line_parser.go:61 +0x193
github.com/derekparker/delve/pkg/dwarf/line.TestDebugLinePrologueParser(0xc0420a80f0)
        D:/go/src/github.com/derekparker/delve/pkg/dwarf/line/line_parser_test.go:61 +0x317
testing.tRunner(0xc0420a80f0, 0x59aad8)
        C:/Go1.9/src/testing/testing.go:754 +0xd7
created by testing.(*T).Run
        C:/Go1.9/src/testing/testing.go:800 +0x31b
FAIL    github.com/derekparker/delve/pkg/dwarf/line     1.215s
```

I've added a bit of debugging for the error message but it's not that helpful:

```
--- FAIL: TestDebugLinePrologueParser (1.08s)
panic: Could not parse string "" from "" [recovered]
        panic: Could not parse string "" from ""

goroutine 35 [running]:
testing.tRunner.func1(0xc042050340)
        c:/go/src/testing/testing.go:622 +0x2a4
panic(0x551300, 0xc04217a2c0)
        c:/go/src/runtime/panic.go:489 +0x2dd
github.com/derekparker/delve/pkg/dwarf/util.ParseString(0xc0421521c0, 0x53278f, 0x62f440, 0xc0421521c0)
        D:/go/src/github.com/derekparker/delve/pkg/dwarf/util/util.go:80 +0x213
github.com/derekparker/delve/pkg/dwarf/line.parseIncludeDirs(0xc0420f8960, 0xc0421521c0)
        D:/go/src/github.com/derekparker/delve/pkg/dwarf/line/line_parser.go:96 +0x36
github.com/derekparker/delve/pkg/dwarf/line.Parse(0xc04217c000, 0xfc00, 0xfc00, 0xc04217c000, 0xfc00, 0xfc00)
        D:/go/src/github.com/derekparker/delve/pkg/dwarf/line/line_parser.go:61 +0x177
github.com/derekparker/delve/pkg/dwarf/line.TestDebugLinePrologueParser(0xc042050340)
        D:/go/src/github.com/derekparker/delve/pkg/dwarf/line/line_parser_test.go:61 +0x358
testing.tRunner(0xc042050340, 0x597348)
        c:/go/src/testing/testing.go:657 +0x9d
created by testing.(*T).Run
        c:/go/src/testing/testing.go:697 +0x2d1
FAIL    github.com/derekparker/delve/pkg/dwarf/line     2.422s
```

It seems that the data source is empty.

--- a/pkg/dwarf/line/line_parser_test.go
+++ b/pkg/dwarf/line/line_parser_test.go
@@ -4,6 +4,7 @@ import (
 	"debug/elf"
 	"debug/macho"
 	"debug/pe"
+	"flag"
 	"os"
 	"os/exec"
 	"path/filepath"
@@ -12,6 +13,14 @@ import (
 
 	"github.com/pkg/profile"
 )
+
+var userTestFile string
+
+func TestMain(m *testing.M) {
+	flag.StringVar(&userTestFile, "user", "", "runs line parsing test on one extra file")
+	flag.Parse()
+	os.Exit(m.Run())
+}
 
 func grabDebugLineSection(p string, t *testing.T) []byte {
 	f, err := os.Open(p)
@@ -28,7 +37,11 @@ func grabDebugLineSection(p string, t *testing.T) []byte {
 
 	pf, err := pe.NewFile(f)
 	if err == nil {
-		data, _ := pf.Section(".debug_line").Data()
+		sec := pf.Section(".debug_line")
+		data, _ := sec.Data()
+		if 0 < sec.VirtualSize && sec.VirtualSize < sec.Size {
+			return data[:sec.VirtualSize]
+		}
 		return data
 	}
 
@@ -45,18 +58,7 @@ const (
 	lineRangeGo18 uint8 = 10
 )
 
-func TestDebugLinePrologueParser(t *testing.T) {
-	// Test against known good values, from readelf --debug-dump=rawline _fixtures/testnextprog
-	p, err := filepath.Abs("../../../_fixtures/testnextprog")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	err = exec.Command("go", "build", "-gcflags=-N -l", "-o", p, p+".go").Run()
-	if err != nil {
-		t.Fatal("Could not compile test file", p, err)
-	}
-	defer os.Remove(p)
+func testDebugLinePrologueParser(p string, t *testing.T) {
 	data := grabDebugLineSection(p, t)
 	debugLines := Parse(data)
 	dbl := debugLines[0]
@@ -111,6 +113,29 @@ func TestDebugLinePrologueParser(t *testing.T) {
 	if !ok {
 		t.Fatal("File names table not parsed correctly")
 	}
+}
+
+func TestUserFile(t *testing.T) {
+	if userTestFile == "" {
+		return
+	}
+	t.Logf("testing %q", userTestFile)
+	testDebugLinePrologueParser(userTestFile, t)
+}
+
+func TestDebugLinePrologueParser(t *testing.T) {
+	// Test against known good values, from readelf --debug-dump=rawline _fixtures/testnextprog
+	p, err := filepath.Abs("../../../_fixtures/testnextprog")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = exec.Command("go", "build", "-gcflags=-N -l", "-o", p, p+".go").Run()
+	if err != nil {
+		t.Fatal("Could not compile test file", p, err)
+	}
+	defer os.Remove(p)
+	testDebugLinePrologueParser(p, t)
 }
 
 func BenchmarkLineParser(b *testing.B) {

